### ISSUE TITLE
Design API to find out current number of logged  in &registered users in vlabs.ac.in

### ISSUE THREAD

**Issue by ksripathi** [openedx-vlead/vlabs-analytics-api-server#10]
vlabs.ac.in is using openedx platform for hosting labs and for single sign in/sign out feature. This issue captures the implementation & documentation to find out the current number of logged in users in openedx platform

**Comment by travula** [comment]
we should also have an api to capture total number of registered users too

**Comment by ksripathi** [comment]
Yes @travula 
We have [API design](https://github.com/vlead/vlabs-analytics-service/blob/analytics-apis/src/runtime/rest/api.org#get-current-logged-in-users-and-total-registered-users-in-vlabsacin) to capture both total number of registered users a and current logged in users

I also implemented an algorithm to get both of them and it is working with all possible test cases and this I would l like to discuss with you  in person

Basically this algorithm process all the =nginx-server-log-files" of openedx platform running machine as input resources

Currently we have =nginx-server-log-files= from the date **2017-June-06** to till date

@sivashankerN 
Can we get =nginx-log-files= before date **2017-June-06** as well this is required to process the total visits and hits of all the labs on openedx platform

**Comment by sivashankerN** [comment]
@ksripathi and @travula  Currently we have nginx logs from 2017-06-08, I do not see logs before this date in edx machine. I am not sure we will get the logs.

I am checking if there is any mechanism in edx platform that deletes excess/after particular period of time logs

**Comment by sivashankerN** [comment]
As i expected there is a mechanism called "log rotation" in edx platform.

It is keeping 4 weeks of logs. it is not just for nginx logs there are other logs of edx platform as well.

**Comment by travula** [comment]
Both the use cases:
1. Total number of registered users
2. Current users logges in

shoudl be retrieved from an open edx API.

If such an API is not present, we shoould pursue using logs for use case 2.
For use case 1, we should never use logs.   If open edx API is not present,
 we should check the database to find out the users present in the database.

Sincerely
Thirumal Ravula

**Comment by ksripathi** [comment]
Yes @travula 
For use case 2 we have chosen logs to compute the total current logged in users and this task is done.
For use case 1 i need  to look into it for existing API if not found we can get them from database as you mentioned

**Comment by travula** [comment]
@sripathi

If we can find out whether the current sessions are also stored in db,  if
that that  works out, then it is straight forward.  We do not need to
depend on the logs.

for use case 2, the strategy could be like this:

a. read the stored value of current sessions or
b. insert a new api call at the same place where the other day, changes
were made for
    setting xframe options.   I think those are the functions for
login/logout I guess , or
c. read from the logs.


Sincerely
Thirumal Ravula

**Comment by ksripathi** [comment]
@travula 

I will look into edx database if there is anything related to session storage and update this issue

**Comment by ksripathi** [comment]
@travula 
For time being we have a python [script](https://github.com/vlead/vlabs-analytics-service/blob/analytics-apis/src/deployment/index.org#script-to-get-openedx-user-analytics) which gets registered, active and inactive users from edx database

Also design and implementation can be found at : https://github.com/vlead/vlabs-analytics-service/blob/analytics-apis/src/runtime/rest/api.org#get-current-logged-in-users-vlabsacin
